### ISSUE TITLE
[BUG] Autologged Classification metrics are not correct

### ISSUE THREAD

**Issue by ddejaco** [mlflow/mlflow#7755]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- Client: 1.x.y
- Tracking server: sqlite:///mlruns.db
- mlflow version 2.1.1

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **Python version**: 3.10.09

### Describe the problem

Sklearn - Classification problem using mlflow.autolog()

- The autologged (during FIT) "training_["recall", "precision", "f1"]_recall" metrics are wrong.
- The values are compared with the correctly computed values using sklearn.metric

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
from sklearn import svm
import mlflow
import pandas
import sklearn
import numpy as np

mlflow.autolog()
mlflow.sklearn.autolog()

mlflow.set_tracking_uri("sqlite:///mlruns.db")
mlflow.set_experiment(experiment_name = "ISSUES Classification")

# These features will lead a model: 1 --> 1, 0 --> 0
XTrain = pandas.DataFrame(np.array([1, 1, 1, 1, 1, 0, 0, 0, 0, 0]).T, columns=['feature'])
yTrain = pandas.DataFrame(np.array([1, 1, 1, 1, 1, 0, 0, 0, 1, 1]).T, columns=['target'])
# confusion matrix (manually counted):
# 5  |  0
# --------
# 2  |  3

with mlflow.start_run() as activeRun:
    
    # classification
    classification = svm.LinearSVC()
    classification = classification.fit(XTrain, yTrain.values.ravel())

    # Compute Training metrics using sklearn.metrics
    m = {}
    kwargsTrain = {"y_true": yTrain, "y_pred": classification.predict(XTrain)}
    m["RECALL_train"] = sklearn.metrics.recall_score(**kwargsTrain)
    m["PREC_train"] = sklearn.metrics.precision_score(**kwargsTrain)
    m["F1_train"] = sklearn.metrics.f1_score(**kwargsTrain)
    
    print("Resulting metrics (sklearn.metrics)")
    for item in m:
        print(item, ": ")
        print(m[item], "\n")
        

finishedRun = mlflow.tracking.MlflowClient().get_run(activeRun.info.run_id)
lMetrics = ["training_recall_score", "training_precision_score", "training_f1_score"]
print("Autologged metrics:")
for item in lMetrics:
    print(item, ":")
    print(finishedRun.data.metrics[item], "\n")
```

### Stack trace

```
No error is returned
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@ddejaco I think the difference comes from `average`:

https://github.com/mlflow/mlflow/blob/9eeeff414d119c7df72b10da3dda8daf000d85ea/mlflow/sklearn/utils.py#L215

**Comment by ddejaco** [comment]
> @ddejaco I think the difference comes from `average`. MLflow uses `weighted` by default:
> 
> https://github.com/mlflow/mlflow/blob/9eeeff414d119c7df72b10da3dda8daf000d85ea/mlflow/sklearn/utils.py#L215
> 
> You can use `mlflow.sklearn.autolog(..., pos_label=1)` for binary classification.

Thank you @harupy for your comment. I see now that by setting **pos_label = 1** the metrics are correct.

I think it is very misleading that the metrics autologged with default arguments are not the same as the sklearn.metrics calculated with default arguments. (the default arguments of e.g. sklearn.metrics.recall_score are **pos_label = 1** and **average = "binary"**)
 Shouldn't the default calculation be the same? (I changed the title accordingly)

**Comment by B-Step62** [comment]
This is expected, documented behavior, not a bug: `mlflow.sklearn.autolog` computes classification metrics with `average="weighted"` by default (switching to `average="binary"` only when you set `pos_label`), whereas `sklearn.metrics` defaults to `average="binary"` for binary problems. Pass `pos_label` to align them. Closing.
